cookies/forms.py

```diff
@@ -279,3 +279,33 @@ class Meta:
 
 class ConceptEntityLinkForm(forms.Form):
     uri = forms.CharField(max_length=255, help_text='Manually enter a ConceptPower URI', label='URI')
+
+
+class UserAddCollectionForm(forms.Form):
+    name = forms.CharField(help_text='Names are unique accross ALL entities in the system.')
+    content_type = forms.ModelChoiceField(**{
+        'queryset': Type.objects.all().order_by('name'),
+        'help_text': 'Types help JARS determine what metadata fields are' \
+                   + ' appropriate for your collection.',
+    })
+    public = forms.BooleanField(**{
+        'required': False,
+        'initial': True,
+        'help_text': u'If checked, this collection will be available to the' \
+                   + u' public. By checking this box you affirm that you have' \
+                   + u' the right to upload and distribute this collection.'
+    })
+    uri = forms.CharField(**{
+        'required': False,
+        'help_text': u'You may (optionally) specifiy an URI for your resource.'\
+                   + u' If you\'re unsure, leave this blank and let JARS' \
+                   + u' generate a URI for you.',
+        'label': 'URI',
+    })
+    resources = forms.MultipleChoiceField(**{
+        'widget': forms.widgets.CheckboxSelectMultiple,
+        'choices': Resource.objects.values_list('id', 'name'),
```

Comment:
@sanjanavsdvn I suspect that this is what is killing the Travis build. One thing that takes some getting used to in Django is that querysets are lazy; they can be instantiated long before resulting in a database call, which is what allows us to do complex and efficient operations. Check out the Django docs for details, but the short story is that calling the ``values_list()`` method is one of the ways that a database call gets triggered. Since these class declarations are executed when the app code is loaded, this database call gets triggered *even before migrations* -- hence the exception that you see in the Travis build. You don't have this problem locally because your database already exists when you're running the tests.

Setting that aside, your intuition here was a good one. To overcome the problem above, however, look for the ``ModelMultipleChoiceField`` -- you can pass a queryset (just like for the ``ModelChoiceField`` that you used above); crucially, it won't get resolved until an instance of the form is rendered, which means that the class declaration itself won't trigger a database error.